src/dbus_api/api.rs

```diff
@@ -80,28 +82,28 @@ fn object_path_to_pair(dbus_context: &DbusContext,
 }
 
 /// Get name for pool from object path
-fn object_path_to_pool_name(dbus_context: &DbusContext,
+fn object_path_to_pool_uuid(dbus_context: &DbusContext,
                             path: &dbus::Path)
-                            -> Result<String, (MessageItem, MessageItem)> {
-    let pool_name = match dbus_context.pools.borrow().get(path) {
+                            -> Result<Uuid, (MessageItem, MessageItem)> {
+    let uuid = match dbus_context.pools.borrow().get(path) {
         Some(pool) => pool.clone(),
         None => {
             let items = code_to_message_items(DbusErrorEnum::INTERNAL_ERROR,
                                               format!("no pool for object path {}", path));
             return Err(items);
         }
     };
-    Ok(pool_name)
+    Ok(uuid)
```

Comment:
The particular construction to which I believe you are referring went away in the macros that were derived from it.